### ISSUE TITLE
[Bug]: Popover doesn't avoid safe areas

### ISSUE THREAD

**Issue by ricardoboss** [minikin/popover#97]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the popover is opened, it may open behind safe areas, that is it may open behind a display cutout or the system toolbar, which is not favourable.

![image](https://github.com/user-attachments/assets/6ec80fda-f810-4d23-a9af-a1e3319adbb3)

### Expected Behavior

I would expect to be able to control whether the popover automatically avoids safe areas when it detects it will intersect with one.

### What operating system are you seeing the problem on?

iOS

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by minikin** [comment]
hey @ricardoboss, have you tried [SafeArea](https://api.flutter.dev/flutter/widgets/SafeArea-class.html)?

**Comment by ricardoboss** [comment]
The issue I have is a bit more complex, so just throwing a `SafeArea` around it wouldn't help (it's in a list that does safe area handling itself anyways).

I tried to create a minimal reproduction but was unable to, so it seems this is a bug in my app, not your lib. Also, I check out the example and there it's handled properly. Closing this issue.

While working on the reproduction, I encountered issues like #90, which seem to happen when calling `showPopover` from a nested widget in a list? Also, when calling `showPopover` from a deeply nested widget it doesn't work.
I suggest you add to the readme that the widget initiating the popover must provide the context as opposed to a parent widget (maybe by wrapping the button opening the popover in a `Builder`).
